contracts/Core/AssetManager.sol

```diff
@@ -64,22 +64,35 @@ contract AssetManager is ACL, AssetStorage {
 
     event CollectionActivityStatus(uint256 id, uint256 epoch, bool active, uint256 timestamp);
 
-    constructor(address parametersAddress) {
+    modifier checkState(uint256 state) {
+        require(state == parameters.getState(), "incorrect state");
+        _;
+    }
+
+    modifier checkStates(uint256 state1, uint256 state2) {
+        require(state1 == parameters.getState() || state2 == parameters.getState(), "incorrect states");
+        _;
+    }
+
+    constructor(address parametersAddress, address blockManagerAddress) {
         parameters = IParameters(parametersAddress);
+        blockManager = IBlockManager(blockManagerAddress);
```

Comment:
Where is the use of blockmanager here
I did not find any call
Please remove if not needed